File: works/tmp/1463.java
import java.util.*;

public class Main 
{
	static int [] d = new int[1000001];
	
	
	public static void main(String[] args) 
	{
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();

		d[0] = 0;
		d[1] = 0;
		
		for(int i = 2; i <= n; i++)
		{
			int num = d[i-1] + 1;
			if(i % 2 == 0) num = Math.min(d[i/2]+1, num);
			if(i % 3 == 0) num = Math.min(d[i/3]+1, num);
			d[i] = num;
		}
		
		System.out.println(d[n]);
	}

}

File: works/tmp/1463.py
d = [0 for _ in range(1000001)]

n = int(input())
d[0] = 0
d[1] = 0

for i in range(2, n + 1):
    num = d[i - 1] + 1
    if i % 2 == 0:
        num = min(num, d[int(i / 2)] + 1)
    if i % 3 == 0:
        num = min(num, d[int(i / 3)] + 1)
    d[i] = num


print(d[n])

